### PR TITLE
cellVdec: replace deprecated ffmpeg function

### PR DESCRIPTION
**Pull request by linkmauve**
`avcodec_decode_video2()` is deprecated, now replaced with `avcodec_send_packet()` and `avcodec_receive_frame()`.

`frame->pkt_pts` is also replaced with `packet.pts`.

### CODE REVIEW

**Comment by linkmauve** [comment]
Ugh, the Windows CI doesn’t have a ffmpeg version from at least March 2016…

File: rpcs3/Emu/Cell/Modules/cellVdec.cpp [hunk 1]
```diff
@@ -255,29 +255,30 @@ struct vdec_context final
...
-						break;
+						av_make_error_string(av_error, AV_ERROR_MAX_STRING_SIZE, ret);
+						fmt::throw_exception("AU queuing error(0x%x): %s" HERE, ret, av_error);
 					}
 
-					if (decode != packet.size)
-					{
-						cellVdec.error("Incorrect AU size (0x%x, decoded 0x%x)", packet.size, decode);
-					}
+					while (true) {
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellVdec.cpp]: Newline for the {

**Comment by Megamouse** [comment]
I think our ffmpeg submodule is on a really old commit on some really old branch iirc

**Comment by tabnk** [comment]
Does it fix interlace video?

**Comment by linkmauve** [comment]
It might because it now handles producing more than one frame per packet, do you have any homebrew I could test it with?

**Comment by tabnk** [comment]
Only know Tekken Tag Tournament HD had interlaced video for the opening video scene.

**Comment by linkmauve** [comment]
I don’t own this game, could you test this PR yourself then?  Or produce a testcase I could use to reproduce your issue?

**Comment by tabnk** [comment]
Ok. Will be testing later. 

Btw, no build from https://ci.appveyor.com/project/rpcs3/rpcs3/builds/28391140/artifacts

**Comment by linkmauve** [comment]
Correct, as it was previously said the ffmpeg submodule is way too old for this new API, I tested against ffmpeg master which is the reason I was getting these warnings in the first place.

**Comment by tabnk** [comment]
> Correct, as it was previously said the ffmpeg submodule is way too old for this new API, I tested against ffmpeg master which is the reason I was getting these warnings in the first place. You’ll have to use a ffmpeg commit from after March 2016.

No sure about using a earlier ffmpeg commit. Can you send/upload a complied Windows build.

**Comment by linkmauve** [comment]
No sorry, I don’t have any Windows computer, nor do I want to go through the hassle to learn how to use one.

**Comment by tabnk** [comment]
Fine. Will test it after this PR is fixed than.

**Comment by ZeeWanderer** [comment]
@linkmauve i managed to build it for windows with some vcpkg triplet trickery, so my test results are probably incorrect.
MGS4 
image

[RPCS3.gz](https://github.com/RPCS3/rpcs3/files/3775021/RPCS3.gz)
Upon encountering interlaced frames.

[bin_win64](https://www.dropbox.com/s/7qajqfg23z99x8h/bin.zip?dl=0)

**Comment by linkmauve** [comment]
Is this a regression?  If you build rpcs3 with this ffmpeg version but without this PR does it also crash that way?

I should have looked just after my commit, this TODO from your log says interlacing still isn’t supported:
```
·U 0:02:22.137654 {PPU[0x100002b] Thread (HLE Video Decoder)} cellVdec TODO: Interlaced frames not supported (0x1)
```

**Comment by 1whatleytay** [comment]
Hey,

I've been implementing avcdec on Vita3K. We needed support for codecs like atrac9 among other things so we decided to fork PPSSPP's implementation and make our own build of ffmpeg. You can find it at https://github.com/Vita3K/ffmpeg-core, it should be up to date with the functions you need if you want to look into it.

It's important to mention I've dropped support for atrac3 and atrac3+ codecs (since we didn't need them- but you probably will), and for any platform that's not x86_64 Windows, Mac or Linux. It dropped the size significantly, but it might be an issue for you. If you want me to reenable anything, I'm ready to rebuild.

**Comment by MsDarkLow** [comment]
Hatsune Miku Project Diva F 1 and 2, Ultimate Marvel Vs Capcom 3 fail to load the pre-rendered intro and crash RPCS3.
Win 10, 8700k, GTX 1070

[HatsuneMiku Project Diva1 PR.zip](https://github.com/RPCS3/rpcs3/files/4014906/HatsuneMiku.Project.Diva1.PR.zip)
[HatsuneMiku Project Diva2 PR.zip](https://github.com/RPCS3/rpcs3/files/4014909/HatsuneMiku.Project.Diva2.PR.zip)
[UMVC3 PR.zip](https://github.com/RPCS3/rpcs3/files/4014910/UMVC3.PR.zip)

On master they play just fine.
[HatsuneMiku Project Diva1 Master.log.gz](https://github.com/RPCS3/rpcs3/files/4014911/HatsuneMiku.Project.Diva1.Master.log.gz)
[HatsuneMiku Project Diva2 Master.log.gz](https://github.com/RPCS3/rpcs3/files/4014914/HatsuneMiku.Project.Diva2.Master.log.gz)
[UMVC3 Master.log.gz](https://github.com/RPCS3/rpcs3/files/4014917/UMVC3.Master.log.gz)

**Comment by Nekotekina** [comment]
I found the bug (frame wasn't allocated properly) and cherry-picked a commit from this PR. I hope @linkmauve  is fine with this.

**Pull request closed without merging**